### PR TITLE
GFORMS-2149 - Move handlebar payload configuration to a separate conf…

### PR DESCRIPTION
**Pull request by DursunGundogan**
…ig file

### CODE REVIEW

File: app/uk/gov/hmrc/gform/sharedmodel/formtemplate/destinations/Destination.scala [hunk 1]
```diff
@@ -411,4 +423,28 @@ object UploadableConditioning {
...
+    val path = Paths.get(url).normalize().toAbsolutePath().toString()
+    val source = Source.fromFile(path)
+    try Some(source.mkString)
+    catch {
+      case ex: Exception =>
+        throw new RuntimeException(
+          s"An error occurs when attempting to read the file from $path in the ${id.id} destination with exception : $ex"
+        )
+    } finally source.close()
+  }
```
Comment (josef-vlach) [inline app/uk/gov/hmrc/gform/sharedmodel/formtemplate/destinations/Destination.scala:441]: What is the idea behind this? I'm not sure what is this for?

Comment (DursunGundogan) [reply app/uk/gov/hmrc/gform/sharedmodel/formtemplate/destinations/Destination.scala:441]: The idea is to read and load the handlebar configuration into the `payload` attribute while posting a form template if payloadUrl is present.

**Review by josef-vlach — CHANGES_REQUESTED** [review]
**External payload needs to be *inlined* into template on upload time....**

When template is uploaded and corresponding external payload do not exists, we need to raise an error. 
Also when external payload is being uploaded, we need to update corresponding formTemplate inline version of it (at the beginning it may not exists.)
If it comes to the question what needs to be first payload or template, the answer is payload.

One indication that this was done right is that gform-frontend would contains exactly 0 changes.

Notes. This only applies for `"type": "handlebarsHttpApi"` destination when `"payload"` property is not defined.

File: app/uk/gov/hmrc/gform/sharedmodel/formtemplate/destinations/Destination.scala [hunk 2]
```diff
@@ -81,10 +79,12 @@ object Destination {
     uri: String,
     method: HttpMethod,
     payload: Option[String],
+    payloadName: Option[String],
```
Comment (josef-vlach) [inline app/uk/gov/hmrc/gform/sharedmodel/formtemplate/destinations/Destination.scala:82]: `payloadName` should be derived. It should equal to `formTemplateId-destinationId`

**Review by josef-vlach — CHANGES_REQUESTED** [review]

**Review by josef-vlach — APPROVED** [review]

**Pull request merged**